src/CSEditor/LightPicker.cpp

```
@@ -0,0 +1,281 @@
+
+	auto* cell = player->GetParentCell();
+	if (!cell)
+		return out;
+
+	const ImVec2 cursor = ImGui::GetMousePos();
+	const ImVec2 display = ImGui::GetIO().DisplaySize;
+	if (display.x <= 0.0f || display.y <= 0.0f)
+		return out;
+
+	static constexpr float kSearchRadius = 5000.0f;          // Skyrim units (~50 m)
+	static constexpr float kScreenThreshSq = 64.0f * 64.0f;  // pixels
```

Cppcheck (2.21.0)

[error] Returning pointer that will be invalid when returning.

(returnDanglingLifetime)